lisa/tools/mono.py

```diff
@@ -0,0 +1,98 @@
+from typing import Any, cast
+
+from lisa.base_tools import Wget
+from lisa.executable import Tool
+from lisa.operating_system import Posix, Ubuntu
+from lisa.tools import Echo
+from lisa.util import UnsupportedDistroException
+
+from .ls import Ls
+
+
+class Mono(Tool):
+    @property
+    def command(self) -> str:
+        return "mono"
+
+    @property
+    def can_install(self) -> bool:
+        return True
+
+    def _initialize(self, *args: Any, **kwargs: Any) -> None:
+        self.nuget_path = "/usr/local/bin/nuget.exe"
+        self._ensure_nuget_installed()
+
+    def _ensure_nuget_installed(self) -> None:
+        ls = self.node.tools[Ls]
+        if not ls.path_exists(self.nuget_path, sudo=True):
+            self.node.tools[Wget].get(
+                url="https://dist.nuget.org/win-x86-commandline/latest/nuget.exe",
+                file_path="/usr/local/bin",
+                filename="nuget.exe",
+                executable=True,
+                sudo=True,
+            )
+
+    def install(self) -> bool:
+        if isinstance(self.node.os, Ubuntu):
+            self._install_mono_on_ubuntu()
+        else:
+            raise UnsupportedDistroException(os=self.node.os)
+        return self._check_exists()
+
+    def _install_mono_on_ubuntu(self) -> None:
+        posix_os: Posix = cast(Posix, self.node.os)
+        required_packages = [
+            "dirmngr",
+            "gnupg",
+            "apt-transport-https",
+            "ca-certificates",
+        ]
+        posix_os.install_packages(required_packages)
+
+        mono_repo_entry = (
+            "deb https://download.mono-project.com/repo/ubuntu stable-bionic main"
```

Comment:
Does it support only bionic? If so, filter out other versions.

Reply:
change into distro codename